### PR TITLE
vpnaas resources

### PR DESCRIPTION
**Pull request by volosatik**
closes CMPT-33972
closes CMPT-33973
closes CMPT-29349
closes CMPT-33974

### CODE REVIEW

File: examples/vpnaas/ipsec_policy/main.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+resource "vkcs_vpnaas_ipsec_policy" "internal_db" {
```
Comment (ftersin) [inline examples/vpnaas/ipsec_policy/main.tf:1]: internal db?

File: examples/vpnaas/endpoint_group/main.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+resource "vkcs_vpnaas_endpoint_group" "allowed_hosts" {
+  name = "allowed-hosts-tf-example"
+  type = "cidr"
```
Comment (ftersin) [inline examples/vpnaas/endpoint_group/main.tf:3]: we need an example for another type of endpoint group as well

Comment (volosatik) [reply examples/vpnaas/endpoint_group/main.tf:3]: added it in subnet subfolder

File: examples/vpnaas/ipsec_policy/main.tf (hunk 1 above)
Comment (volosatik) [reply examples/vpnaas/ipsec_policy/main.tf]: renamed

File: templates/vpnaas/resources/vkcs_vpnaas_endpoint_group.md.tmpl [hunk 3]
```diff
@@ -11,7 +11,11 @@ description: |-
 {{ .Description }}
 
 ## Example Usage
+### Use manual cidr endpoints
```
Comment (ftersin) [inline templates/vpnaas/resources/vkcs_vpnaas_endpoint_group.md.tmpl:14]: Remote endpoint group

File: templates/vpnaas/resources/vkcs_vpnaas_endpoint_group.md.tmpl [hunk 4]
```diff
@@ -11,7 +11,11 @@ description: |-
...
 {{tffile .ExampleFile}}
+
+### Use subnet endpoints
```
Comment (ftersin) [inline templates/vpnaas/resources/vkcs_vpnaas_endpoint_group.md.tmpl:17]: Local (private) endpoint group

**Review by ftersin — APPROVED** [review]

**Pull request merged**